tests/whenHandlingMultipleSubscriptions.js

```diff
@@ -49,20 +51,27 @@ describe('When handling multiple subscriptions', function () {
         subscription1 = new Subscription(host1, port1, uri1, 1.2);
         subscription2 = new Subscription(host2, port2, uri2, 1.2);
         let sub1, sub2;
-        subscription1.on('subscribed', function(payload) {
-            sid1 = payload.sid;
-            if (sub2) {
-                done();
-            } else {
-                sub1 = true;
+
+        async.parallel({
+            subscription1: function(callback) {
+                subscription1.on('subscribed', function(payload) {
+                    console.log('subscribed')
+                    sid1 = payload.sid;
+
+                    callback(null, true);
+                });
+            },
+            subscription2: function(callback) {
+                subscription2.on('subscribed', function(payload) {
+                    console.log('subscribed')
+                    sid2 = payload.sid;
+
+                    callback(null, true);
+                });
             }
-        });
-        subscription2.on('subscribed', function(payload) {
-            sid2 = payload.sid;
-            if (sub1) {
+        }, function(err, results) {
```

Comment:
Nice! Really good idea to use `async` instead of my shoddy implementation before!